interpreter/src/main/scala/pl/touk/nussknacker/engine/testing/ModelDefinitionBuilder.scala

```diff
@@ -54,19 +58,29 @@ final case class ModelDefinitionBuilder(
       name: String,
       returnType: Option[TypingResult],
       componentSpecificData: CustomComponentSpecificData,
-      params: Parameter*
+      componentId: Option[ComponentId],
```

Comment:
Please move componentId on the and, add the default `None` and remove all duplications of `componentId = None`. Thanks to that we won't spread this edge case through the all places where it is not important.

Reply:
I've tried to add a default case, but scala isn't too happy with it, even though it should be easy for it to differentiate between type `Parameter` and type `Option[ComponentId]`

- *-parameter must come last
- Parameter section with *-parameter is not allowed to have default arguments

Reply:
ok i have an idea:

```
  def withCustom(
      name: String,
      returnType: Option[TypingResult],
      componentSpecificData: CustomComponentSpecificData,
      componentId: Option[ComponentId] = None,
  ): ModelDefinitionBuilder =
    withCustom(name, returnType, componentSpecificData, componentId, Seq.empty: _*)

  def withCustom(
      name: String,
      returnType: Option[TypingResult],
      componentSpecificData: CustomComponentSpecificData,
      componentId: Option[ComponentId],
      params: Parameter*
  ): ModelDefinitionBuilder =
    wrapWithNotDisabledCustomComponentDefinition(
      params.toList,
      returnType,
      componentSpecificData
    ).map(withComponent(name, _, componentId)).getOrElse(this)

```